platform/ios/src/MGLFaux3DUserLocationAnnotationView.m

```diff
@@ -245,7 +245,9 @@ - (void)drawDot
             _oldHeadingAccuracy = -1;
         }
 
-        if ( ! _headingIndicatorLayer && headingAccuracy)
+        bool validHeadingAccuracy = headingAccuracy > 0 && headingAccuracy < kCLLocationAccuracyNearestTenMeters;
```

Comment:
This line compares `headingAccuracy` (measured in degrees) to `kCLLocationAccuracyNearestTenMeters` (measured in meters).